pkg/router.go

```diff
@@ -158,12 +163,69 @@ func PointWrite(m *nmodule.Module, r *router.Request) ([]byte, error) {
 		return nil, err
 	}
 
-	// TODO: Encode PointWriter (pw)
+	serialData := endec.NewSerialData()
+	endec.SetPositionalData(serialData, true)
+	endec.SetRequestData(serialData, true)
+	msgId, _ := endec.GenerateRandomId()
+	endec.SetMessageId(serialData, msgId)
+	endec.UpdateBitPositionsForHeaderByte(serialData)
+
 	log.Infof("Point Address UUID >>>>>>> %s ", nstring.DerefString(pnt.AddressUUID))
 
-	// TODO: Encrypt the encoded data by using encrypter.Encrypt method
+	// If the PointWriter struct has fields, you can also print them individually
+	// Assuming pw has fields like `Name` and `Value`
+	for _, value := range *pw.Priority {
+		if value != nil {
+			floatValue := *value
+			pointDataType, err := strconv.Atoi(pnt.DataType) // As DataType is store as string
+			if err != nil {
+				return nil, err
+			}
+			if endec.MetaDataKey(pointDataType) == endec.MDK_UINT_8 {
+				endec.EncodeData(serialData, uint8(floatValue), endec.MetaDataKey(pointDataType), uint8(*pnt.AddressID))
+			} else if endec.MetaDataKey(pointDataType) == endec.MDK_UINT_16 {
+				endec.EncodeData(serialData, uint16(floatValue), endec.MetaDataKey(pointDataType), uint8(*pnt.AddressID))
+			} else if endec.MetaDataKey(pointDataType) == endec.MDK_UINT_32 {
+				endec.EncodeData(serialData, uint32(floatValue), endec.MetaDataKey(pointDataType), uint8(*pnt.AddressID))
+			} else if endec.MetaDataKey(pointDataType) == endec.MDK_UINT_64 {
+				endec.EncodeData(serialData, uint64(floatValue), endec.MetaDataKey(pointDataType), uint8(*pnt.AddressID))
+			} else if endec.MetaDataKey(pointDataType) == endec.MDK_INT_8 {
+				endec.EncodeData(serialData, int8(floatValue), endec.MetaDataKey(pointDataType), uint8(*pnt.AddressID))
+			} else if endec.MetaDataKey(pointDataType) == endec.MDK_INT_16 {
+				endec.EncodeData(serialData, int16(floatValue), endec.MetaDataKey(pointDataType), uint8(*pnt.AddressID))
+			} else if endec.MetaDataKey(pointDataType) == endec.MDK_INT_32 {
+				endec.EncodeData(serialData, int32(floatValue), endec.MetaDataKey(pointDataType), uint8(*pnt.AddressID))
+			} else if endec.MetaDataKey(pointDataType) == endec.MDK_INT_64 {
+				endec.EncodeData(serialData, int64(floatValue), endec.MetaDataKey(pointDataType), uint8(*pnt.AddressID))
+			} else {
+				endec.EncodeData(serialData, floatValue, endec.MetaDataKey(pointDataType), uint8(*pnt.AddressID))
+			}
```

Comment:
be careful de-referencing `*pnt.AddressID`
@maharjanaman can you please add that util function to this. if nil, return error instead maybe